pkgs/development/python-modules/temporalio/default.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  buildPythonPackage,
+  cargo,
+  fetchFromGitHub,
+  poetry-core,
+  protobuf,
+  python-dateutil,
+  rustc,
+  rustPlatform,
+  setuptools,
+  setuptools-rust,
+  types-protobuf,
+  typing-extensions,
+}:
+
+buildPythonPackage rec {
+  pname = "temporalio";
+  version = "1.8.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "temporalio";
+    repo = "sdk-python";
+    rev = "refs/tags/${version}";
+    hash = "sha256-xJIb+U3hV6DDvfbfY20U/4rE2z0XzaESkehtcvG6m50=";
+    fetchSubmodules = true;
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    sourceRoot = "${src.name}/${cargoRoot}";
+    hash = "sha256-+A226LfxqmUA/CMXwx8osgIRJhfKtB8KcCSBkbbXOHk=";
+  };
+
+  cargoRoot = "temporalio/bridge";
+
+  patches = [ ./protoc-wheel-removal.diff ];
+
+  build-system = [
+    poetry-core
+    setuptools
+    setuptools-rust
+  ];
+
+  dependencies = [
+    protobuf
+    python-dateutil
+    types-protobuf
+  ];
+
+  nativeBuildInputs = [
+    cargo
+    rustPlatform.cargoSetupHook
+    rustc
+  ];
+
+  propagatedBuildInputs = [ typing-extensions ];
+
+  pythonImportsCheck = [
+    "temporalio"
+    "temporalio.client"
+    "temporalio.worker"
+  ];
+
+  meta = {
+    description = "Temporal Python SDK";
+    homepage = "https://temporal.io/";
```

```suggestion
    homepage = "https://temporal.io/";
    changelog = "https://github.com/temporalio/sdk-python/releases/tag/${version}";
```
Adding meta.changelog is useful for future updates.